Function Documentation with Fixed Signatures and Event Links (#557)
commerce-docs/microsite-commerce-storefront: src/content/docs/dropins/cart/functions.mdx

  - Cleaning Storybook imports and metadata
  - Combining all functions into a single documentation page
  
  The template and script must be kept in sync.
  
  HEADING HIERARCHY:
  H1: "DROPIN_NAME functions" (from title)
  H2: Individual function names
  H3: Examples, Events, Returns (subsection headings)
  Content flow: [Signature code block] → [Parameters table] → Examples → Events → Returns
*/}

import TableWrapper from '@components/TableWrapper.astro';
import Link from '@components/Link.astro';
import { Aside } from '@astrojs/starlight/components';

The Cart drop-in provides API functions that enable you to programmatically control behavior, fetch data, and integrate with Adobe Commerce backend services.

<div style="background-color: var(--sl-color-blue-low); border-left: 4px solid var(--sl-color-blue); padding: 0.75rem 1rem; border-radius: 0.25rem; margin: 1rem 0;">
<strong>Version: 1.5.1</strong>
</div>

{/* AUTO-GENERATED CONTENT - Do not edit below this line */}

<TableWrapper nowrap={[0]}>

| Function | Description |
| --- | --- |
| [`addProductsToCart`](#addproductstocart) | The addProductsToCart function adds products to a cart. |

> | [`applyCouponsToCart`](#applycouponstocart) | A function that applies or replaces one or more coupons to the cart.. |

Reviewer: **issue (trivial):** Double period

| [`resetCart`](#resetcart) | This function resets the cart drop-in. |
| [`setGiftOptionsOnCart`](#setgiftoptionsoncart) | setGiftOptionsOnCart is a function that sets gift options on the cart. |
| [`updateProductsFromCart`](#updateproductsfromcart) | The updateProductsFromCart function updates cart items by either changing the quantity or removing and adding an item in one step. |

</TableWrapper>

## addProductsToCart

The `addProductsToCart` function adds products to a cart. You must supply a `sku` and `quantity`for each product. The other parameters are specified for complex product types. The function calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/add-products/" text="addProductsToCart" /> mutation.

```ts
const addProductsToCart = async (
  items: { sku: string; parentSku?: string; quantity: number; optionsUIDs?: string[]; enteredOptions?: { uid: string; value: string }[]; customFields?: Record<string, any>; }[]
): Promise<CartModel | null>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
|---|---|---|---|

> | `sku` | `string` | Yes | The product identifier (SKU) to add to the cart. For configurable products (like a shirt available in multiple colors and sizes), use the child product SKU that represents the specific variant selected by the customer (e.g., `MS09-M-Blue` for a medium blue shirt). |

Reviewer 2: No Latin abbreviations

```suggestion
| `sku` | `string` | Yes | The product identifier (SKU) to add to the cart. For configurable products (like a shirt available in multiple colors and sizes), use the child product SKU that represents the specific variant selected by the customer (`MS09-M-Blue` for a medium blue shirt). |
```

I don't think all of this detail is needed.

| `parentSku` | `string` | No | For configurable products, this is the SKU of the parent (base) product. For example, if adding a specific variant like `MS09-M-Blue` (child SKU), the `parentSku` would be `MS09` (the base configurable product). This helps Commerce track the relationship between the variant and its parent product. |

> | `quantity` | `number` | Yes | The number of items to add to the cart. For example, `1` to add a single item, or `3` to add three units of the product. This value must be a positive number. |

Reviewer 2: ```suggestion
| `quantity` | `number` | Yes | The number of the selected item to add to the cart. For example, `1` to add a single item, or `3` to add three units of the product. This value must be a positive number. |
```

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## applyCouponsToCart

A function that applies or replaces one or more coupons to the cart.

```ts
const applyCouponsToCart = async (
  couponCodes: string[],
  type: ApplyCouponsStrategy
): Promise<CartModel | null>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
|---|---|---|---|
| `couponCodes` | `string[]` | Yes | An array of coupon codes to apply to the cart. |
| `type` | `ApplyCouponsStrategy` | Yes | The strategy for applying coupons. See [`ApplyCouponsStrategy`](#applycouponsstrategy). |

</TableWrapper>

### Events

Emits the [`cart/updated`](/dropins/cart/events/#cartupdated-emits-and-listens) and [`cart/data`](/dropins/cart/events/#cartdata-emits-and-listens) events with the updated cart information after applying the coupons.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## applyGiftCardToCart

> The applyGiftCardToCart function is used to apply a gift card to the current shopping cart. It takes the gift card code as an argument and updates the cart with the applied gift card.

Reviewer 2: ```suggestion
The `applyGiftCardToCart` function is used to apply a gift card to the current shopping cart. It takes the gift card code as an argument and updates the cart with the applied gift card.
```

The `getCartData` function is mainly used internally by the `initializeCart()` and `refreshCart()` functions. If you need detailed information about the current user's shopping cart, a more optimal approach is to listen for `cart/data` or `cart/updated` events so that you do not need to make another network call.

```ts
const getCartData = async (): Promise<CartModel | null>
```

### Events

Does not emit any drop-in events.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## getEstimatedTotals

A function that returns estimated totals for cart based on an address. It takes an `address` parameter.

```ts
const getEstimatedTotals = async (
  address: EstimateAddressShippingInput
): Promise<CartModel | null>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
|---|---|---|---|
| `address` | `EstimateAddressShippingInput` | Yes | The shipping address used to calculate estimated cart totals, taxes, and shipping costs. See [`EstimateAddressShippingInput`](#estimateaddressshippinginput). |

</TableWrapper>

### Events

Does not emit any drop-in events.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## getEstimateShipping

> The `getEstimateShipping` function returns the first available shipping method and its estimated cost, based on the provided address. The function calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql-api/index.html#mutation-estimateShippingMethods" text="estimateShippingMethods" /> mutation. Note: This function returns raw GraphQL data. For a transformed `ShippingMethod` object, listen to the `shipping/estimate` event instead.

Reviewer 2: It returns an array. I suspect the whole Note is fiction.

```suggestion
The `getEstimateShipping` function returns a list of available shipping methods and itheir estimated costs, based on the provided address. The function calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql-api/index.html#mutation-estimateShippingMethods" text="estimateShippingMethods" /> mutation. Note: This function returns raw GraphQL data. For a transformed `ShippingMethod` object, listen to the `shipping/estimate` event instead.
```

### Returns

Returns `any | null`.

---

## getStoreConfig

The `getStoreConfig` function returns information about a store's configuration. The function calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/store/queries/store-config/" text="storeConfig" /> query.

```ts
const getStoreConfig = async (): Promise<StoreConfigModel | null>
```

### Events

Does not emit any drop-in events.

### Returns

Returns [`StoreConfigModel`](#storeconfigmodel) or `null`.

---

## initializeCart

The `initializeCart` function initializes a guest or customer cart. This function is automatically called during the initialize phase of a dropin's lifecycle. You do not need to call this manually.

```ts
const initializeCart = async (): Promise<CartModel | null>
```

### Events

Emits the [`cart/initialized`](/dropins/cart/events/#cartinitialized-emits), [`cart/data`](/dropins/cart/events/#cartdata-emits-and-listens), and [`cart/merged`](/dropins/cart/events/#cartmerged-emits-and-listens) events. The event payload contains data about the address and shipping method.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## publishShoppingCartViewEvent

> Publishes a shopping cart view event to the ACDL. This function sets the shopping cart context and triggers a `SHOPPING_CART_VIEW` event on the Adobe Client Data Layer, typically used when a cart page loads.

Reviewer 2: ```suggestion
Publishes a shopping cart view event to the Adobe Client Data Layer (ACDL). This function sets the shopping cart context and triggers a `SHOPPING_CART_VIEW` event on the ACDL, typically used when a cart page loads.
```

```ts
const publishShoppingCartViewEvent = async (): any
```

### Events

Does not emit any drop-in events.

> Publishes the `SHOPPING_CART_VIEW` event to the Adobe Client Data Layer (ACDL) with the current cart context.

Reviewer 2: ```suggestion
Publishes the `SHOPPING_CART_VIEW` event to the ACDL with the current cart context.
```

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## removeGiftCardFromCart

This function removes a single gift card from the cart. It function calls the <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/remove-giftcard/" text="removeGiftCardFromCart" /> mutation.

```ts
const removeGiftCardFromCart = async (
  giftCardCode: string
): Promise<CartModel | null>
```

<TableWrapper nowrap={[0, 1]}>

| Parameter | Type | Req? | Description |
|---|---|---|---|
| `giftCardCode` | `string` | Yes | Defines the gift card code to remove. |

</TableWrapper>

### Events

Emits the [`cart/updated`](/dropins/cart/events/#cartupdated-emits-and-listens) and [`cart/data`](/dropins/cart/events/#cartdata-emits-and-listens) events.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## resetCart

This function resets the cart drop-in. As a result, the cart ID is set to null and the authenticated status is set to false.

```ts
const resetCart = async (): Promise<CartModel | null>
```

### Events

Does not emit any drop-in events.

### Returns

Returns [`CartModel`](#cartmodel) or `null`.

---

## setGiftOptionsOnCart

> <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/set-gift-options/" text="setGiftOptionsOnCart" /> is a function that sets gift options on the cart. It takes a `giftOptions` parameter.

Reviewer 2: ```suggestion
The `setGiftOptionsOnCart` function sets options such as gift receipts and wrapping. It calls the  <Link href="https://developer.adobe.com/commerce/webapi/graphql/schema/cart/mutations/set-gift-options/" text="setGiftOptionsOnCart" /> mutation.
```
